zerver/lib/actions.py

```diff
@@ -760,12 +760,31 @@ def do_scrub_realm(realm: Realm) -> None:
         user.delivery_email = scrubbed_email
         user.save(update_fields=["full_name", "email", "delivery_email"])
 
+    internal_realm = get_realm(settings.SYSTEM_BOT_REALM)
+    all_recipient_ids_in_realm = (
+        list(Stream.objects.filter(realm=realm).values_list('recipient_id', flat=True)) +
+        list(UserProfile.objects.filter(realm=realm).values_list('recipient_id', flat=True)) +
+        list(
+            Subscription.objects.filter(recipient__type=Recipient.HUDDLE, user_profile__realm=realm)
+            .values_list('recipient_id', flat=True)
+        )
```

Comment:
I'm a bit reluctant on the Huddle front, just because the constraint is "any recipient of the huddle", which feels a bit risky to me (the other things are really clearly scoped as data only in this realm).  Is it actually possible to send a cross-realm huddle message? I though it was not.

Reply:
I'm not aware of any other way, but you can using `internal_send_huddle_message` like in the test in this commit.
```
        internal_send_huddle_message(
            zulip,
            notification_bot,
            [othello.email, iago.email],
            "test"
        )
```

this only works for cross-realm bots though, and the target users have to be within my realm (I fiddled around with the function to verify). So we should be good on the " clearly scoped as data only in this realm)" front.